test/lit/ctor-eval/array_new_data.wast

```diff
@@ -2,10 +2,11 @@
 ;; RUN: wasm-ctor-eval %s --ctors=test --kept-exports=test --quiet -all -S -o - | filecheck %s
 
 (module
+ (type $"[i8]" (array i8))
+
  ;; CHECK:      (type $0 (func))
 
  ;; CHECK:      (type $[i8] (array i8))
- (type $[i8] (array i8))
```

Comment:
This moved away because the names no longer match, I guess? Should we update both at once?

Reply:
Yeah, and it looks like the follow-up hasn't fixed it fully. Updating both at once makes sense.

Reply:
@kripken PTAL. The commits on this PR should be separately reviewable, but we can land them all together to avoid intermediate test regressions.